go/gilded-rose_test.go

```diff
@@ -1,15 +1,151 @@
+func Print_Items(items []*Item){
+	fmt.Println("name, sellIn, quality")
+	for i := 0; i < len(items); i++ {
+		fmt.Println(items[i])
+	}
+	fmt.Println("")
+}
 func Test_Foo(t *testing.T) {
 	var items = []*Item{
-		&Item{"foo", 0, 0},
+		&Item{"+5 Dexterity Vest", 10, 20},
+		&Item{"Aged Brie", 2, 0},
+		&Item{"Elixir of the Mongoose", 5, 7},
+		&Item{"Sulfuras, Hand of Ragnaros", 0, 80},
+		&Item{"Sulfuras, Hand of Ragnaros", -1, 80},
+		&Item{"Backstage passes to a TAFKAL80ETC concert", 15, 20},
+		&Item{"Backstage passes to a TAFKAL80ETC concert", 10, 49},
+		&Item{"Backstage passes to a TAFKAL80ETC concert", 5, 49},
+		&Item{"Conjured Mana Cake", 15, 40}, // <-- :O
+		&Item{"Conjured Mana Cake", 3, 46}, // <-- :O
+	}
+
+	days := 12
+
+	for day := 0; day < days; day++ {
+		fmt.Printf("-------- day %d --------\n", day)
+		Print_Items(items)
+		UpdateQuality(items)
+	}
+	fmt.Printf("-------- LAST day %d --------\n", days)
+	Print_Items(items)
+
+	//+5 Dexterity Vest
+	if items[0].name != "+5 Dexterity Vest" {
+		t.Errorf("Name: Expected %s but got %s ", "+5 Dexterity Vest", items[0].name)
+	}
+	if items[0].sellIn != -2 {
+		t.Errorf("Sellin: Expected %s but got %s ", "-2", strconv.Itoa(items[0].sellIn))
+	}
+	if items[0].quality != 6 {
+		t.Errorf("Quality: Expected %s but got %s ", "6", strconv.Itoa(items[0].quality))
+	}
+
+	//Aged Brie -10 22}
+	if items[1].name != "Aged Brie" {
+		t.Errorf("Name: Expected %s but got %s ", "Aged Brie", items[1].name)
+	}
+	if items[1].sellIn != -10 {
+		t.Errorf("Sellin: Expected %s but got %s ", "-10", strconv.Itoa(items[1].sellIn))
+	}
+	if items[1].quality != 22 {
+		t.Errorf("Quality: Expected %s but got %s ", "22", strconv.Itoa(items[1].quality))
+	}
+
+	//Elixir of the Mongoose -7 0}
+	if items[2].name != "Elixir of the Mongoose" {
+		t.Errorf("Name: Expected %s but got %s ", "Elixir of the Mongoose", items[2].name)
+	}
+	if items[2].sellIn != -7 {
+		t.Errorf("Sellin: Expected %s but got %s ", "-7", strconv.Itoa(items[2].sellIn))
+	}
+	if items[2].quality != 0 {
+		t.Errorf("Quality: Expected %s but got %s ", "0", strconv.Itoa(items[2].quality))
+	}
+
+	//Sulfuras, Hand of Ragnaros 0 80}
+	if items[3].name != "Sulfuras, Hand of Ragnaros" {
+		t.Errorf("Name: Expected %s but got %s ", "Sulfuras, Hand of Ragnaros", items[3].name)
+	}
+	if items[3].sellIn != 0 {
+		t.Errorf("Sellin: Expected %s but got %s ", "0", strconv.Itoa(items[3].sellIn))
+	}
+	if items[3].quality != 80 {
+		t.Errorf("Quality: Expected %s but got %s ", "80", strconv.Itoa(items[3].quality))
 	}
 
-	UpdateQuality(items)
+	//Sulfuras, Hand of Ragnaros -1 80}
+	if items[4].name != "Sulfuras, Hand of Ragnaros" {
+		t.Errorf("Name: Expected %s but got %s ", "Sulfuras, Hand of Ragnaros", items[4].name)
+	}
+	if items[4].sellIn != -1 {
+		t.Errorf("Sellin: Expected %s but got %s ", "-1", strconv.Itoa(items[4].sellIn))
+	}
+	if items[4].quality != 80 {
+		t.Errorf("Quality: Expected %s but got %s ", "80", strconv.Itoa(items[4].quality))
+	}
```

Comment:
It's always nice to have context, these expectations make no sense if I haven't read the rules of why these are so. You could instead have a text description to explain this rule in plain text and using a table test, and have that printed in the error for the test.